google-cloud-datastore/src/main/java/com/google/cloud/datastore/AggregationQuery.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.datastore;
+
+import static com.google.cloud.datastore.AggregationQuery.Mode.GQL;
+import static com.google.cloud.datastore.AggregationQuery.Mode.STRUCTURED;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.cloud.datastore.aggregation.Aggregation;
+import com.google.cloud.datastore.aggregation.AggregationBuilder;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AggregationQuery extends Query<AggregationResults> {
+
+  private List<Aggregation> aggregations;
+  private StructuredQuery<?> nestedStructuredQuery;
+  private final Mode mode;
+  private GqlQuery<?> nestedGqlQuery;
+
+  AggregationQuery(String namespace, List<Aggregation> aggregations,
+      StructuredQuery<?> nestedQuery) {
+    super(checkNotNull(namespace));
+    checkArgument(!aggregations.isEmpty(),
+        "At least one aggregation is required for an aggregation query to run");
+    this.aggregations = aggregations;
+    this.nestedStructuredQuery = nestedQuery;
+    this.mode = STRUCTURED;
+  }
+
+  AggregationQuery(String namespace, GqlQuery<?> gqlQuery) {
+    super(checkNotNull(namespace));
+    this.nestedGqlQuery = gqlQuery;
+    this.mode = GQL;
+  }
+
+  public List<Aggregation> getAggregations() {
+    return aggregations;
+  }
+
+  public StructuredQuery<?> getNestedStructuredQuery() {
+    return nestedStructuredQuery;
+  }
+
+  public GqlQuery<?> getNestedGqlQuery() {
+    return nestedGqlQuery;
+  }
+
+  public Mode getMode() {
+    return mode;
+  }
+
+  public static class Builder {
```

Comment:
Hi @gforgurups, In a way we can do that. But we need AggregationQuery's `Builder` due to following reasons:

- End user is accustomed to using builder (ex: Query.newEntityQueryBuilder, Query.newKeyQueryBuilder etc.), and we created a new `Query.newAggregationQueryBuilder` to access this builder class. If we expose AggregationQuery class directly we will be breaking the pattern established in their mind.
- If we see from code perspective, we have followed the existing convention and maintain the consistency with other Query classes by exposing this builder.

These two reasons are good enough for us to keep having this Builder, let me know if you think otherwise.